templates/mustache-templates/teamsBot.js.mustache

```diff
@@ -6,51 +6,54 @@ const { TeamsActivityHandler, CardFactory{{#IS_BOT}}, ActionTypes{{/IS_BOT}}} =
 
 class {{#IS_BOT}}TeamsBot{{/IS_BOT}}{{^IS_BOT}}MessageExtensionBot{{/IS_BOT}} extends TeamsActivityHandler {
   {{#IS_BOT}}
-  /**
-   * @param {ConversationState} conversationState
-   * @param {UserState} userState
-   * @param {Dialog} dialog
-   */
-  constructor(conversationState, userState, dialog) {
+ 
+  constructor() {
     super();
-    if (!conversationState) {
-      throw new Error("[TeamsBot]: Missing parameter. conversationState is required");
-    }
-    if (!userState) {
-      throw new Error("[TeamsBot]: Missing parameter. userState is required");
-    }
-    if (!dialog) {
-      throw new Error("[TeamsBot]: Missing parameter. dialog is required");
-    }
-    this.conversationState = conversationState;
-    this.userState = userState;
-    this.dialog = dialog;
-    this.dialogState = this.conversationState.createProperty("DialogState");
 
-    this.onMessage(async (context, next) => {
-      console.log("Running dialog with Message Activity.");
+    // record the likeCount
+    this.likeCountObj = { likeCount: 0 };
 
-      // Run the Dialog with the new message Activity.
-      await this.dialog.run(context, this.dialogState);
+    this.onMessage(async (context, next) => {
+      console.log("Running with Message Activity.");
+
+      let txt = context.activity.text;
+      if (!txt) {
+        if (context.activity.value) {
+          // Means this activity came from a invoke action of Adaptive Card
+          this.likeCountObj.likeCount++;
```

Comment:
If there are several action in adaptive card, does this mean the count will be increased when user click either one of them?

Reply:
resolved by change it to onAdaptiveCardInvoke, and recognize the invoke by invokeValue.action.verb === "userlike"

Reply:
resolved, recognized by verb